### PR TITLE
Chisels now work on all atoms that can be smoothed

### PR DESCRIPTION
**Pull request by ZephyrTFA**
Also makes it so that chisels are smarter in how they function

### CODE REVIEW

**Review by MarkSuckerberg — APPROVED** [review]

**Comment by ZephyrTFA** [comment]
DNM until I improve logic for toggling smooth flags

**Comment by ZephyrTFA** [comment]
Alright, its more intelligent now, although I havent tested the new code at all so TM it first 🤡

**Comment by ZephyrTFA** [comment]
Test merge looks good, still would rather not merge my own PR. @francinum

**Comment by PrefabQuasar** [comment]
removing smoothing on any non-turf object doesnt properly update smoothing, and toggling smoothing on open turfs that lack smoothing(such as floor tiles) causes error tiles to appear
image

**Comment by ZephyrTFA** [comment]
> removing smoothing on any non-turf object doesnt properly update smoothing, and toggling smoothing on open turfs that lack smoothing(such as floor tiles) causes error tiles to appear image

Removing soothing causes atoms to no longer smooth yes. Did you word this incorrectly? 
And the error tiles is probably from an invalid base iconstate, I'll look into it

**Comment by PrefabQuasar** [comment]
> > removing smoothing on any non-turf object doesnt properly update smoothing, and toggling smoothing on open turfs that lack smoothing(such as floor tiles) causes error tiles to appear image
> 
> Removing soothing causes atoms to no longer smooth yes. Did you word this incorrectly? And the error tiles is probably from an invalid base iconstate, I'll look into it

sorry yes, meant to clarify that removing smoothing from non-turf atoms is permanent, and no amount of further toggling will cause them to re-smooth

**Comment by ZephyrTFA** [comment]
Alright, that I can look into. Thanks

**Comment by ZephyrTFA** [comment]
@MarkSuckerberg 🦎

**Comment by MarkSuckerberg** [comment]
> @MarkSuckerberg 🦎 

A

File: code/game/objects/items/tools/chisel.dm [hunk 1]
```diff
@@ -1,59 +1,129 @@
...
+	else
+		if(!atom_supports_diagonal(target))
+			to_chat(user, "<span class='warning'>\the [src] makes a tart buzz. \the [target] doesn't appear to support smoothed corners.</span>")
+			return
+		smooth_atom_diagonal(target, user)
+	return
 
 /obj/item/chisel/attack_self(mob/user)
 	toggling_smooth = !toggling_smooth
 	to_chat(user, "<span class='notice'>\the [src] is now set to reform the [(toggling_smooth ? "smoothness" : "corners")] of walls</span>")
```
Comment (MarkSuckerberg) [inline code/game/objects/items/tools/chisel.dm]:
```suggestion
	to_chat(user, "<span class='notice'>\the [src] is now set to reform the [(toggling_smooth ? "smoothness" : "corners")] of objects.</span>")
```

**Comment by PrefabQuasar** [comment]
noticed recently that this causes chisels to be impossible to place onto tables, racks, lockers, etc without throwing them onto it, only gives you the prompt that the object cannot be smoothed, would probably be solved with a call to parent in there somewhere

**Comment by ZephyrTFA** [comment]
> noticed recently that this causes chisels to be impossible to place onto tables, racks, lockers, etc without throwing them onto it, only gives you the prompt that the object cannot be smoothed, would probably be solved with a call to parent in there somewhere

Done

**Comment by PrefabQuasar** [comment]
only other issue i noticed was that wall corners seemed to get reset to intitial any time ships moved, but i suspect that would be VASTLY outside of the scope of this PR, so i'll make an issue report about it later

**Comment by ZephyrTFA** [comment]
> only other issue i noticed was that wall corners seemed to get reset to intitial any time ships moved, but i suspect that would be VASTLY outside of the scope of this PR, so i'll make an issue report about it later

That's an issue with shuttle move not copying over critical vars for turfs

**Pull request merged**